hal/lib/athena/Notifier.cpp

```diff
@@ -102,35 +126,27 @@ void* initializeNotifier(void (*process)(uint64_t, void*), void* param,
 
   std::lock_guard<priority_recursive_mutex> sync(notifierMutex);
   // create notifier structure and add to list
-  Notifier* notifier = new Notifier();
-  notifier->prev = nullptr;
+  std::shared_ptr<Notifier> notifier = std::make_shared<Notifier>(Notifier());
```

Comment:
The inner construction isn't necessary.. just `std::make_shared<Notifier>()` should be sufficient (and is more efficient).